src/Paramore.Brighter/CommandProcessor.cs

```diff
@@ -493,13 +453,41 @@ await handlerChain.First().HandleAsync(command, cancellationToken)
         /// Please note that this call will not participate in any ambient Transactions, if you wish to have the outbox participate in a Transaction please Use Deposit,
         /// and then after you have committed your transaction use ClearOutbox
         /// </summary>
-        /// <typeparam name="T"></typeparam>
         /// <param name="request">The request.</param>
+        /// <param name="transactionProvider">The transaction provider</param>
+        /// <typeparam name="TRequest">The type of request</typeparam>
+        /// <typeparam name="TTransaction">The type of transaction used by the Outbox</typeparam>
         /// <exception cref="System.ArgumentOutOfRangeException"></exception>
-        public void Post<T>(T request) where T : class, IRequest
+        public void Post<TRequest, TTransaction>(TRequest request, IAmABoxTransactionProvider<TTransaction> transactionProvider) where TRequest : class, IRequest
         {
-            ClearOutbox(DepositPost(request, null));
+            ClearOutbox(DepositPost(request, transactionProvider));
```

Comment:
If you write the IRequest into a transaction then one two following things should happen
  - when we go to clear if we don't select inside of the txn then there will be no message to dispatch
  - if we do select inside of the transaction then if we roll back the txn the message will already be sent

Reply:
Will review. I was trying to: provide consistent interfaces for transactions rather than force casting, in the handler; eliminate generics from the command processor interface that could result from that. It may be that it can't be null because we have to infer a transaction type, but it may be that we can just use the new default CommittableTransaction type, which can effectively act as a "do nothing"